Common/cpp/SharedItems/Shareable.cpp

```diff
@@ -114,7 +181,7 @@ jsi::Value ShareableValue::toJSValue(jsi::Runtime &rt) {
     case StringType:
       return jsi::Value(rt, jsi::String::createFromAscii(rt, stringValue));
     case ObjectType:
-      return createHost(rt, frozenObject);
+      return createFrozenWrapper(this, rt, frozenObject);
```

Comment:
would it make sense to use the same approach with remote object and mutable object. They won't support enumeration now as well and it seem like we can follow a very similar pattern there too and get rid of `createHost` completely

Reply:
I tried but with the current "adapt" approach it's hard for remoteObj and we still want to control access to mutable.